app/decorators/application_decorator.rb

```diff
@@ -1,5 +1,33 @@
+# frozen_string_literal: true
+
 class ApplicationDecorator < Draper::Decorator
   delegate_all
-  
+
   self.include_root_in_json = true
+
+  def api_selector_api_link
+    raise NotImplementedError, 'Should be implemented in subclasses [BackendApiDecorator, ServiceApiDecorator]'
```

Comment:
No need for the message IMHO. Superclasses should know nothing about subclasses. It's OK to make it abstract, but names or implementation details of the subclasses are too much. It's meant to get outdated.